### ISSUE TITLE
Predicting only the NOTHING class

### ISSUE THREAD

**Issue by ninja18** [asingh33/CNNGestureRecognizer#25]
When I try to predict using the pre trained model it only predicts the NOTHING class everytime with 99.9% accuracy I am running theano in CPU.

**Comment by asingh33** [comment]
Accuracy of the program will highly depend on the surrounding lighting condition, how is your backdrop etc. I have added a new filter mode - background subtraction and so far its the most suitable filter for such kind of usage. Try it, to activate it use 'x'

**Comment by 458087641** [comment]
I got this problem too, the prediction always be nothing. but sometimes if the hand get to the edge of the window, the prediction become "ok", however, the "nothing" still occupy the large amount. i did open the background filter. By the way, I was running the program on windows system.

**Comment by asingh33** [comment]
Please try with the latest updated project. I have added a 3rd filter mode - background subtraction. Activated by 'x' key.

**Comment by 458087641** [comment]
> Please try with the latest updated project. I have added a 3rd filter mode - background subtraction. Activated by 'x' key.

I used the latest project, and open the filter, but still only got "nothing"

**Comment by asingh33** [comment]
@458087641 
Ok something is definitely wrong. Accuracy cant go that bad. Can you please elaborate more on the steps you are taking while initiating the script ?

**Comment by vivek6449** [comment]
I am also having the same issue. Is there anyone who solved it? It's only predicting NOTHING class. After downloading weights from google drive, I replaced it with GitHub weights and then ran trackgesture.py file.

**Comment by asingh33** [comment]
@vivek6449 , @458087641 @ninja18  
Which OS are you all using ? Is it Linux ?
BTW I have only tested this project on Mac and Windows so far.

**Comment by foreign** [comment]
I try on windows. The same problem occurs always NOTHING. 
I aloso activated by 'x' key.

**Comment by SantoshBajantri** [comment]
Hey thanks for the repo! Its working quite well in windows OS(Not all time but sometimes when background is removed properly) but its completely not working in Ubuntu. Having the same issue of showing 'NOTHING' always. Please help me to increase the accuracy in Ubuntu. And I don't have any idea how it could be a OS issue? 

Thank You!

**Comment by asingh33** [comment]
Are you guys initiating the actual prediction process by pressing the 'g' key once all the cam windows are launched ?

'g - Toggle Prediction Mode'

**Comment by SantoshBajantri** [comment]
Yes.. First I'm pressing 'x' key to remove the background and then pressing 'g'. Its behaving same on both OS now. Confidence of "NOTHING"  is more always, not predicting other gesture options.

**Comment by asingh33** [comment]
The trained file provided with the project is not 100% perfect, ideal word would be its over fit as its just based on just single user (which is me :) ) and two modes only. 
Overall accuracy may increase if you try with more simpler background such as plane wall, with enough lighting. 
Alternatively, accuracy can increase if more training data is available. 
This framework also has option to let you capture and create your own database as well. Thats how I did. 
My advise would be if the accuracy for you is low then it would be good to train it for your scenario as well.

**Comment by asingh33** [comment]
Just retrain the model.
I have recently found a new observation that same pretrained weight file doesn't work across the platform. The default weight files I had earlier provided with the repo was on Windows. So it won't work as it is for macOS or Linux.
Today I have provided links to two macOS and Windows specific weight files.
Pull the latest code and try these weight file or retrain. It will work.
